gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1947,3 +1929,91 @@ def test_unsupported_parameter_rest_asyncio():
     )
 
 {% endmacro %}
+
+{# is_rest_unsupported_method renders:
+ # 'True' if transport is async REST.
+ # 'True' if transport is sync REST and method is a client streaming method.
+ # 'False' otherwise.
+#}
+{# NOTE: We will keep updating this method as we add support for methods in async REST. #}
+{% macro is_rest_unsupported_method(method, is_async) %}
+{%- if method.client_streaming or is_async -%}
+{{'True'}}
+{%- else -%}
+{{'False'}}
+{%- endif -%}
+{% endmacro %}
+
+{# rest_transport_specific_tests generates all the rest specific tests for both
+# sync and async transport. It refactors the test cases in macro::rest_required_tests
+# and should replace it once we cover all the test cases in it here.
+#}
+{% macro rest_transport_specific_tests(service, transport, is_async) %}
+{% if 'rest' in transport %}
```

Comment:
1. I've filed an issue and linked it to follow up with adding gRPC tests here as suggested.
2. I've updated the macro name to be more generic.

I don't think we need to raise a generation error for gRPC tests since it would add unnecessary toil in the test suite (i.e. this suite shouldn't be run against gRPC transport which is why we're skipping for it).

I think It's fine to skip generation for it as long as we have a tracking bug to include gRPC tests.

Reply:
Thanks!

> I don't think we need to raise a generation error for gRPC tests since it would add unnecessary toil in the test suite (i.e. this suite shouldn't be run against gRPC transport which is why we're skipping for it).

The idea is to just have a fail-safe to ensure we're indeed skipping it. It protects us from our own future programming errors. I've found this defensive practice valuable.

Not a blocker for this PR, but I still say it's a nice to have.

Reply:
I agree that it's nice to have a fail-safe. Let's sync on this offline and we can address this as a follow up if needed.